app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.to_a.delete_if { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@johnpmitsch I prefer delete_if because there is no bang alternative that may introduce subtle bugs.

2.2.2 :001 > [1,2,3,4].reject { |x| x > 10 }
 => [1, 2, 3, 4] 

2.2.2 :002 > [1,2,3,4].reject! { |x| x > 10  }                                                                                                                              
 => nil 
2.2.2 :003 > [1,2,3,4].delete_if { |x| x > 10 }
 => [1, 2, 3, 4]